docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+    npred = source_irf_model.npred
+    dnde = source_irf_model.dnde
+    flux = source_irf_model.flux
+
+
+Improve `SourceModels` class
+----------------------------
+The existing `CompoundSkyModel` is a nice very generic abstraction to support
+any kind of arithmetics between `SkyModel` objects, but the number of use
+cases for other operators, except for "+" is very limited and
+can always be achieved by implementing a custom model. I/O and component handling
+of this hierarchical model component structure is intrinsically difficult. For
+this reason we propose to first support and improve the existing `SourceModels` that
+implements an easier to handle flat hierarchy for model components. Support for
+arbitrary model arithmetics can be introduced, if needed, after Gammapy v1.0. We
+propose to remove the `CompoundSourceModel` and reimplement the `+` operator using
+the `SourceModels` class.
+
+.. code-block::
+
+    from gammapy.cube import SourceModel, SourceModels
+
+    component_1 = SourceModel()
+    component_2 = SourceModel()
+
+    total_model = component_1 + component_2
+    assert isinstance(total_model, SourceModels)
+
+    # becomes quivalent to
+
+    total_model = SourceModels([component_1, component_2])
+
+Possibly remove the `SpectralCompoundModel` for consistency.
+
+
+Introduction of model name attributes
+-------------------------------------
+The `SourceModel` class should be improved to support model component names. This
+way the `SourceModels` object can be improved to access model components by name,
+as well as the XML serialization (which supports names) becomes complete.
+The following example should work:
+
+.. code-block::
+
+    from gammapy.cube import SourceModel, SourceModels
+
+    component_1 = SourceModel(name="source_1")
+    component_2 = SourceModel(name="source_2")
+
+    total_model = SourceModels([component_1, component_2])
+
+    total_model["source_1"].parameters["index"].value = 2.3
+
+    # or alternatively??
+
+    total_model.parameters["source_1.index"].value = 2.3
+
+    # delete a model component in place
+    del model["source_2"]
+
+This is also simplifies the parameter access in the fitting back-end, because
+parameter names become unique. E.g. no need for cryptic `par_00X_` parameter names
+in the minuit backend, which simplifies debugging and interaction by the user with
+methods such as `Fit.likelihood_profile()` or `Fit.confidence()`, where parameter
+identifiers must be given.
+
+
+Improve and implement model I/O
+-------------------------------
+Serialization of models is an important feature for standard analyses. In
+`gammapy.utils.serialization` there is a prototype implementation of XML
+serialization for `SourceModels`, We propose to fix this existing XML I/O
+prototype so that the following easily works:
+
+.. code-block::
+
+    from gammapy.cube import SourceModels
+
+    model = SourceModels.read("model.xml")
```

Comment:
`model` -> `models` for the variable name here and in the following examples as well.
(always use `models` for a `SourceModels` object, no?)